homeassistant/components/flipr/coordinator.py

```diff
@@ -6,39 +6,46 @@
 from flipr_api import FliprAPIRestClient
 from flipr_api.exceptions import FliprError
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import CONF_FLIPR_ID
-
 _LOGGER = logging.getLogger(__name__)
 
 
-class FliprDataUpdateCoordinator(DataUpdateCoordinator):
-    """Class to hold Flipr data retrieval."""
+class BaseDataUpdateCoordinator(DataUpdateCoordinator):
+    """Parent class to hold Flipr and Hub data retrieval."""
+
+    config_entry: ConfigEntry
 
-    def __init__(self, hass, entry):
+    def __init__(
+        self, hass: HomeAssistant, entry: ConfigEntry, flipr_or_hub_id: str
+    ) -> None:
         """Initialize."""
         username = entry.data[CONF_EMAIL]
         password = entry.data[CONF_PASSWORD]
-        self.flipr_id = entry.data[CONF_FLIPR_ID]
+        self.device_id = flipr_or_hub_id
 
-        # Establishes the connection.
         self.client = FliprAPIRestClient(username, password)
-        self.entry = entry
+        self.config_entry = entry
```

Comment:
```suggestion
```

Reply:
This is automatically set in the coordinator

Reply:
ok